File: source/backend/python/application/domain/model/money.py
# お金に関するドメインモデル
from __future__ import annotations

class Money:
    def __init__(self, amount:int):
        self.amount = amount
    
    def is_positive(self) -> bool:
        return self.amount >= 0
    
    def is_negative(self) -> bool:
        return self.amount < 0
    
    @staticmethod
    def add(money_a: Money, money_b: Money) -> Money:
        return Money(money_a.amount + money_b.amount)
    
    @staticmethod
    def subtract(money_a: Money, money_b: Money) -> Money:
            return Money(money_a.amount - money_b.amount)

File: source/backend/typescript/application/domain/model/Money.ts
/**
 * お金に関するドメインモデル
 */

export class Money {
    readonly amount: number;

    constructor(amount: number) {
        this.amount = amount
    }

    public isPositive(): boolean {
        return this.amount >= 0
    }

    public isNegative(): boolean {
        return this.amount < 0
    }

    public static add(moneyA: Money, moneyB: Money): Money {
        return new Money(moneyA.amount + moneyB.amount);
    }

    public static subtract(moneyA: Money, moneyB: Money): Money {
        return new Money(moneyA.amount - moneyB.amount);
    }
}